### PR TITLE
test: update e2e-aws to use worker groups

### PR DESCRIPTION
**Pull request by rsmitty**
This PR updates the e2e-aws flow to use worker groups and an updated control plane variable setup.

### CODE REVIEW

File: .drone.jsonnet [hunk 1]
```diff
@@ -674,7 +674,7 @@ local e2e_aws_prepare = Step(
 local tf_apply = TriggerDownstream(
   'tf-apply',
   'e2e-talos-tf-apply',
-  ['siderolabs/contrib@main'],
+  ['siderolabs/contrib@worker-groups'],
```
Comment (rsmitty) [inline .drone.jsonnet:677]: update back to main before merging.

**Comment by frezbo** [comment]
I guess contrib PR first?>

**Comment by rsmitty** [comment]
> I guess contrib PR first?>

Indeed. See https://github.com/siderolabs/contrib/pull/26

**Review by frezbo — APPROVED** [review]

**Comment by rsmitty** [comment]
Gonna run through e2e once more here to double check everything is working as expected.

**Pull request merged**